src/modules/menu/components/search/SearchBar.vue

```diff
@@ -20,16 +20,21 @@ const searchQuery = computed(() => store.state.search.query)
 const hasResults = computed(() => store.state.search.results.length > 0)
 const isPhoneMode = computed(() => store.getters.isPhoneMode)
 
-watch(hasResults, (newValue) => {
-    // if an entry has been selected from the list, do not show the list again
-    // because the list has been hidden by onEntrySelected. Also if the search bar is pristine (not
-    // yet modified by the user) we don't want to show the result (e.g. at startup with a swisssearch
-    // query param)
-    if (!selectedEntry.value && !isPristine.value) {
-        log.debug(`Search has result changed to ${newValue}, change the show result to ${newValue}`)
-        showResults.value = newValue
-    }
-})
+watch(
+    hasResults,
+    (newValue) => {
+        // if an entry has been selected from the list, do not show the list again
+        // because the list has been hidden by onEntrySelected.
+        if (!selectedEntry.value) {
+            // if (!selectedEntry.value) {
+            log.debug(
+                `Search has result changed to ${newValue}, change the show result to ${newValue}`
+            )
+            showResults.value = newValue
+        }
+    },
+    { immediate: true }
```

Comment:
Why do you need the immediate ? would be good to comment it

Reply:
i added a comment